extendr-api/src/robj/mod.rs

```diff
@@ -322,18 +322,18 @@ pub trait Types: GetSexp {
 impl Types for Robj {}
 
 impl Robj {
-    /// Is this object is an NA scalar?
+    /// Is this object is an `NA` scalar?
```

Comment:
I would elaborate that this function still operates on vectors, it's just that it checks for length-1 vectors. ie it checks for scalars in the R sense, but not the rust sense.

Reply:
Alright; Given an `Robj` that is a vector `c(NA, NA)` it will say `false` though. Thus I need it to be crystal clear on that. Especially since it is used everywhere in the Rust conversion code.

Reply:
You haven't explained this in doc comment though?

Reply:
Perhaps compare this to the vectorized `|` and `&` vs scalar comparison of `||` and `&&`. 

> `is_na_scalar()` is intended to be used on individual elements of a vector. It is _not_ vectorized and will return `false` if the vector has a length greater than 1.